### PR TITLE
fix limit

### PR DESCRIPTION
**Pull request by duckladydinh**
Hallo,

I like this super high quality library and use it a lot. But I recognized some errors with array limits which  rendered the count_primes unusable and I would like to fix it in this pull request.

Thanks.

### CODE REVIEW

File: count_primes.cpp [hunk 1]
```diff
@@ -46,5 +46,5 @@ void init_count_primes() {
         prime[j] = false;
 
   REP(i, MAX) if (prime[i]) P.push_back(i);
-  FOR(i, 1, MAX) prec[i] = prec[i-1] + prime[i];
+  FOR(i, 1, MAX - 1) prec[i] = prec[i-1] + prime[i];
```
Comment (stjepang) [inline count_primes.cpp]: I don't understand why this chnage is needed -- is it wrong to compute `prec[MAX - 1]`?

File: count_primes.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ llint rec(llint N, int K) {
   const int LIM = 250;
   static int memo[LIM*LIM][LIM];
 
-  bool ok = N < LIM*LIM;
+  bool ok = N < LIM*LIM && K < LIM;
```
Comment (stjepang) [inline count_primes.cpp]: This makes a lot of sense, good catch! 👍

File: count_primes.cpp (hunk 1 above)
Comment (duckladydinh) [reply count_primes.cpp]: The max indexes of the prec and prime array are MAX - 1. I am not sure how older compilers work but clang++8 and g++8 make it a runtime error.

Comment (duckladydinh) [reply count_primes.cpp]: Ah so I thought it to be <= and always used it like that. If so, that is not a problem. Thanks.

**Pull request merged**